### ISSUE TITLE
Allow first part of a path to be a path param

### ISSUE THREAD

**Issue by chris-kahn** [type-route/type-route#13]
It seems like type-route doesn't allow the first part of a path definition to be a parameter? Is that intended behaviour or a bug? For example I'd like to define routes that start with a locale, for example `/en-us/test` or `/en-gb/test`. 

Here's a quick snippet to test with:

```typescript
const testRouter = createRouter({
    testA: defineRoute({ locale: "path.param.string" }, (p) => `/${p.locale}/test`),
    testB: defineRoute({ locale: "path.param.string" }, (p) => `/first/${p.locale}/test`)
})
console.log("A match", testRouter.routes.testA.match({ pathName: "/en-us/test" })) // false
console.log("B match", testRouter.routes.testB.match({ pathName: "/first/en-us/test" })) // { locale: 'en-us' }
```

The `testA` route fails to match, but I expected it to match with `{ locale: 'en-us' }`.

**Comment by chris-kahn** [comment]
I opened #14 which I think can fix this.

**Comment by bradenhs** [comment]
Thanks! Just released your fix in version `0.3.2`.
